pkg/testing/ess/provisioner.go

```diff
@@ -0,0 +1,163 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package ess
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"golang.org/x/sync/errgroup"
+
+	"github.com/elastic/elastic-agent/pkg/testing/runner"
+)
+
+// ProvisionerConfig is the configuration for the ESS provisioner.
+type ProvisionerConfig struct {
+	Identifier string
+	APIKey     string
+	Region     string
+}
+
+// Validate returns an error if the information is invalid.
+func (c *ProvisionerConfig) Validate() error {
+	if c.Identifier == "" {
+		return errors.New("field Identifier must be set")
+	}
+	if c.APIKey == "" {
+		return errors.New("field APIKey must be set")
+	}
+	if c.Region == "" {
+		return errors.New("field Region must be set")
+	}
+	return nil
+}
+
+type provisioner struct {
+	logger runner.Logger
```

Comment:
Because then each package is not coupled by a common interface definition and any changes to the interface in `runner` package do not affect the `ess` package ? Usually interfaces are always defined at the receiver's end by specifying only the actual functions used... if it's really always the same for all the packages everywhere then we can have a common interface for the whole project with the big disclaimer that any changes would affect the whole codebase